src/ElectronNET.API/Common/ApiEventManager.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using ElectronNET.API;
+using ElectronNET.API.Entities;
+using Newtonsoft.Json.Linq;
+
+namespace ElectronNET.Common;
+
+internal static class ApiEventManager
+{
+    internal static void AddEvent(string eventName, object id, Action callback, Action value, string suffix = "")
```

Comment:
The major problem here is that callback will always be null when entering this function, because `callback += value;` will not modify the delegate of the caller, only the local one.
This means effectively, that once a second event registration is made, the first one will stop working.

Reply:
The new PR with integration tests confirms that. I added two tests for multiple-event registrations (which are failling):

https://github.com/ElectronNET/Electron.NET/pull/913/files#diff-111ec18c5df437a8bb282fdb7d7fb28a17a2e09f7bc06b475f0abbef3d654dd5R21-R62